lib/bugsnag/report.rb

```diff
@@ -153,6 +162,26 @@ def ignore!
       @should_ignore = true
     end
 
+    ##
+    # Generates a summary to be attached as a breadcrumb
+    #
+    # @return [Hash] a Hash containing the report's error class, error message, and severity
+    def summary
+      # Guard against the exceptions array being removed/changed or emptied here
+      if !exceptions.is_a?(Array) || exceptions.first.nil?
```

Comment:
I think this would be easier to read if you changed the condition to `exceptions.respond_to?(:first) && exceptions.first` and put the known error class case first.